### PR TITLE
New variable from type field of AppBinding + Fix RoleBinding name conflict with KubeDB

### PR DESCRIPTION
**Pull request by hossainemruz**
Requirements:
- [x] Merge https://github.com/stashed/stash/pull/847
- [x] Rebase against master

Task:
- [x] Use database type as prefix of Repository and BackupConfiguration
- [x] Test manually
- [ ] Update docs

### CODE REVIEW

**Comment by tamalsaha** [comment]
```
t = postgres
g = ""
r = postgres

GVR
GVK

TARGET_APP_TYPE     := kubedb.com/postgres
TARGET_APP_GROUP    := kubedb.com
TARGET_APP_RESOURCE := postgres

prefix: ${NS}/postgres/{name}
```

**Comment by tamalsaha** [comment]
@hossainemruz , can you fix the conflicts?

File: pkg/controller/appbindings.go [hunk 1]
```diff
@@ -78,20 +80,23 @@ func (c *StashController) applyBackupAnnotationLogicForAppBinding(ab *appCatalog
...
+		inputs[apis.TargetAppType] = string(ab.Spec.Type)
 
 		err = resolve.ResolveBackupBlueprint(backupBlueprint, inputs)
 		if err != nil {
 			return err
 		}
 
 		// ensure Repository crd
-		err = c.ensureRepository(backupBlueprint, targetRef)
+		err = c.ensureRepository(backupBlueprint, targetRef, targetAppResource)
```
Comment (tamalsaha) [inline pkg/controller/appbindings.go]: Why are we using the `Resource` from AppBinding here?

File: pkg/controller/appbindings.go [hunk 2]
```diff
@@ -58,6 +58,8 @@ func (c *StashController) applyBackupAnnotationLogicForAppBinding(ab *appCatalog
 		return fmt.Errorf("failed to create reference of %s %s/%s. Reason: %v", ab.Kind, ab.Namespace, ab.Name, err)
 	}
 
+	targetAppGroup, targetAppResource := ab.GroupResource()
```
Comment (tamalsaha) [inline pkg/controller/appbindings.go]: These are optional & it should be ok for these to be empty. Just not usable in that case.

File: pkg/controller/appbindings.go (hunk 1 above)
Comment (hossainemruz) [reply pkg/controller/appbindings.go]: removed

**Pull request merged**